Add setup development environment script file (#44)
l3af-project/l3af-arch: dev_environment/README.md

The L3AF dev environment is a [Vagrant](https://www.vagrantup.com/) virtual
machine environment that allows users to develop, test, or just try out L3AF.

> The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](setup_dev_env_linux_vm.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with Arm or x86 CPUs, Windows WSL and Azure Linux VMs. 

Reviewer: ```suggestion
The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers. To setup the l3afd development environment on a standalone Linux virtual machine please see [setup_dev_env_linux_vm.md](setup_dev_env_linux_vm.md). Use this method for MAC/PC with Arm CPUs, Windows WSL, or cloud provider Linux VMs. 
```

Author: Changed accordingly

> As an alternative to Vagrant, the l3af development environment can be setup on a standalone Linux virtual machine. Please see [setup_dev_env_linux_vm.md](setup_dev_env_linux_vm.md) for setup instructions. Use this method for MAC/PC with ARM CPUs, Windows WSL, or cloud provider Linux VMs.

Reviewer 2: ```suggestion
As an alternative to Vagrant, the l3af development environment can be set up on a standalone Linux virtual machine. Please see [setup_dev_env_linux_vm.md](setup_dev_env_linux_vm.md) for setup instructions. Use this method for MAC/PC with Arm CPUs, Windows WSL, or cloud provider Linux VMs.
```

Author: Changed accordingly

> The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/l3af-project/l3af-arch/blob/main/dev_environment/setup_dev_env_linux_vm.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with Arm or x86 CPUs, Windows WSL and Azure Linux VMs. 

Author: Change to the relative link

> The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/charleskbliu0/l3af-arch/blob/main/dev_environment/Setup_Dev_Env.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with ARM or x86 CPUs, Windows WSL and cloud based Linux VM (i.e. Azure Linux VM). 

Reviewer 2: ```suggestion
The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/charleskbliu0/l3af-arch/blob/main/dev_environment/Setup_Dev_Env.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with Arm or x86 CPUs, Windows WSL and cloud based Linux VM (i.e., Azure Linux VM). 
```

> The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/charleskbliu0/l3af-arch/blob/main/dev_environment/Setup_Dev_Env.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with ARM or x86 CPUs, Windows WSL and cloud based Linux VM (i.e. Azure Linux VM). 

Author: renamed the file "setup_dev_env_linux_vm.md".  Replace  "charleskbliu0" with "l3af-project" for the link.

> 

Reviewer 2: setup_dev_env_linux_vm.md is not referenced in this README.md and should be.  
I think this PR would obsolete most of PR #42, correct?  That PR does update README.md.

Author: Merged the PR#42 section of running rate limiting on the Azure VM.

> The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/l3af-project/l3af-arch/blob/main/dev_environment/setup_dev_env_linux_vm.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with Arm or x86 CPUs, Windows WSL and cloud based Linux VM (i.e. Azure Linux VM). 

Reviewer 2: ```suggestion
The Vagrant virtual machine environment only works for non Apple M1/M2 CPU based MAC computers.  A separate [shell script](https://github.com/l3af-project/l3af-arch/blob/main/dev_environment/setup_dev_env_linux_vm.md) can set up the Linux virtual machine L3AF development environment for MAC/PC with Arm or x86 CPUs, Windows WSL and Azure Linux VMs.
```

Author: changed accordingly

* To see the eBPF program metrics, browse to `http://localhost:33000` on the
  host and login to Grafana with the default username and password of `admin`.
  After logging in you will be able to view the preconfigured dashboards.

# Trying out L3AF on an Azure VM (Ubuntu 20.04.4 LTS)
[setup_dev_env_linux_vm.md](setup_dev_env_linux_vm.md) illustrates host prerequisites and installation instructions for trying out L3AF on an Azure VM running an Ubuntu 20.04.4 LTS server.

> ## Trying out your L3AFD Server

Reviewer: This section is almost the same as what I have above minus the screenshots.  If we want to combine them that would be better than having two different sections.  We can also just drop this entirely and keep what is above.

Author: I combined them and showed the screenshots at the right place.

> The [setup_dev_env_linux_vm.md](https://github.com/l3af-project/l3af-arch/blob/main/dev_environment/setup_dev_env_linux_vm.md) illustrates host prerequisites and installation instructions for trying out L3AF on an Azure VM running an Ubuntu 20.04.4 LTS server.

Reviewer 2: ```suggestion
[setup_dev_env_linux_vm.md](setup_dev_env_linux_vm.md) illustrates host prerequisites and installation instructions for trying out L3AF on an Azure VM running an Ubuntu 20.04.4 LTS server.
```

Author: Changed accordingly
